### ISSUE TITLE
[BUG] Bayes is applied to encrypted messages

### ISSUE THREAD

**Issue by heraklit256** [rspamd/rspamd#3394]
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Read about bug reporting in general: https://rspamd.com/doc/faq.html#how-to-report-bugs-found-in-rspamd
    * Enabled relevant debugging logs: https://rspamd.com/doc/faq.html#how-to-debug-some-module-in-rspamd 
    * Checked the FAQs about Core files in case of fatal crash: https://rspamd.com/doc/faq.html#how-to-figure-out-why-rspamd-process-crashed
    * Tried ASAN package and obtained the ASAN report (if possible): https://rspamd.com/doc/faq.html#asan-builds 
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=%E2%9C%93&q=is%3Aissue+user%3Arspamd
    * Checked that there is not already an experimental package or master branch

**Describe the bug**
Although correctly detecting an PGP encrypted message as encrypted (`ENCRYPTED_PGP`), `rspamd` still applies the Bayes filter on it (`BAYES_HAM`), thus scanning - and potentially learning - encrypted garbage:
```
X-Spamd-Result: default: False [-4.28 / 11.00];
	 ARC_NA(0.00)[];
	 R_DKIM_ALLOW(-0.20)[x];
	 FROM_HAS_DN(0.00)[];
	 DWL_DNSWL_MED(-2.00)[x:dkim];
	 TO_MATCH_ENVRCPT_ALL(0.00)[];
	 IP_REPUTATION_HAM(-0.35)[asn: x(-0.34), country: DE(-0.01), ip: x(0.00)];
	 MIME_GOOD(-0.20)[multipart/encrypted];
	 HAS_ATTACHMENT(0.00)[];
	 TO_DN_NONE(0.00)[];
	 RCVD_DKIM_ARC_DNSWL_MED(-0.50)[];
	 RCPT_COUNT_ONE(0.00)[1];
	 ARC_SIGNED(0.00)[i=1];
	 R_SPF_ALLOW(-0.20)[+ip6:x];
	 MX_GOOD(-0.01)[];
	 RCVD_IN_DNSWL_MED(-0.20)[x:from];
	 DMARC_NA(0.00)[x];
	 DKIM_TRACE(0.00)[x:+];
	 RCVD_COUNT_ZERO(0.00)[0];
	 FROM_EQ_ENVFROM(0.00)[];
	 MIME_TRACE(0.00)[0:+,1:~,2:~];
	 ASN(0.00)[asn:x, ipnet:x, country:DE];
	 BAYES_HAM(-0.12)[66.67%];
	 ENCRYPTED_PGP(-0.50)[]
```

### Steps to Reproduce

1. Set up `rspamd` and get the Bayes filter working (some learning might be required).
2. Scan an PGP (S/MIME should work as well) encrypted message which is correctly identified as such.
3. The `BAYES_HAM` or `BAYES_SPAM` symbol is applied to that message.

**Expected behavior**
Since it does not make sense to classify or learn encrypted data, my expectation is to skip Bayes classification as soon as a message is recognised as being encrypted by `rspamd`.

### Versions

```
[user@machine ~]$ rspamd --version
Rspamd daemon version 2.5
```
```
[user@machine ~]$ cat /etc/os-release 
NAME=FreeBSD
VERSION=12.1--HBSD
VERSION_ID=12.1
ID=freebsd
ANSI_COLOR="0;31"
PRETTY_NAME="FreeBSD 12.1--HBSD"
CPE_NAME=cpe:/o:freebsd:freebsd:12.1
HOME_URL=https://FreeBSD.org/
BUG_REPORT_URL=https://bugs.FreeBSD.org/
```

### Additional Information

Thank you for having a look at this.

**Comment by heraklit256** [comment]
IMHO this issue is still valid.

**Comment by heraklit256** [comment]
This issue appears to has been fixes in rspamd 2.6:
```
2020-10-16 09:57:16 #29063(rspamd_proxy) <d0d372>; proxy; bayes_classify: skipped classification as there are no text tokens. Total tokens: 16
```
